test/fhirtypes/ElementDefinition.fixInstanceDefinition.test.ts

```diff
@@ -0,0 +1,169 @@
+import path from 'path';
+import { loadFromPath } from '../../src/fhirdefs/load';
+import { FHIRDefinitions } from '../../src/fhirdefs/FHIRDefinitions';
+import { StructureDefinition, InstanceDefinition } from '../../src/fhirtypes';
+import { TestFisher } from '../testhelpers';
+
+describe('ElementDefinition', () => {
+  let defs: FHIRDefinitions;
+  let patient: StructureDefinition;
+  let observation: StructureDefinition;
+  let fisher: TestFisher;
+
+  beforeAll(() => {
+    defs = new FHIRDefinitions();
+    loadFromPath(
+      path.join(__dirname, '..', 'testhelpers', 'testdefs', 'package'),
+      'testPackage',
+      defs
+    );
+    fisher = new TestFisher().withFHIR(defs);
+  });
+  beforeEach(() => {
+    patient = fisher.fishForStructureDefinition('Patient');
+    observation = fisher.fishForStructureDefinition('Observation');
+  });
+
+  describe('#fixInstance', () => {
+    it('should fix an allowed type of an instance', () => {
+      const addressInstance = new InstanceDefinition();
+      addressInstance._instanceMeta.name = 'USPostalAddress';
+      addressInstance._instanceMeta.sdType = 'Address';
+      addressInstance._instanceMeta.usage = 'Inline';
+      addressInstance.country = 'US';
+      addressInstance.type = 'postal';
```

Comment:
Technically, `type` is a `code`, so I think this should be:
```ts
addressInstance.type = new FshCode('postal'); // * type = #postal
```

Reply:
Since this is an `InstanceDefinition`, I think that the code would be represented just as the key/value pair with a string for 'postal'. I think if this was setting a rule on an Instance, then we'd use FshCode, but since this is the representation after it's exported, I think it might just be a string. I could be wrong though, so let me know if you still disagree.

Reply:
Ah, yes.  You are right.  (Of course!)